NitroxClient/MonoBehaviours/Gui/Chat/PlayerChatInputField.cs

```diff
@@ -1,132 +1,63 @@
-
-        private IMultiplayerSession session;
-
-        /// <summary>
-        ///     Gets or sets the chat enabled. Can fail to enable chat if another input field in-game already has focus.
-        /// </summary>
-        public bool ChatEnabled
+        private void Awake()
         {
-            get
-            {
-                return chatEnabled;
-            }
-            set
-            {
-                if (!CanEnable())
-                {
-                    return;
-                }
-                // Reuse UWE logic to lock player in-place.
-                if (value)
-                {
-                    inputGroup.OnSelect(true);
-                }
-                else
-                {
-                    inputGroup.OnDeselect();
-                }
-
-                chatEnabled = value;
-            }
+            playerChatManager = NitroxServiceLocator.LocateService<PlayerChatManager>();
         }
 
-        public PlayerChat Manager { get; set; }
-
-        /// <summary>
-        ///     Gets true if no other (UWE) input field has focus right now.`
-        /// </summary>
-        /// <returns>True if the chat input field can be enabled without interfering with another input field in-game.</returns>
-        public bool CanEnable()
+        public void OnSelect(BaseEventData eventData)
         {
-            return FPSInputModule.current.lastGroup == null;
+            playerChatManager.SelectChat();
+            selected = true;
+            ResetTimer();
         }
 
-        public void Awake()
+        public void OnDeselect(BaseEventData eventData)
         {
-            session = NitroxServiceLocator.LocateService<IMultiplayerSession>();
-            inputGroup = gameObject.AddComponent<uGUI_InputGroup>();
+            selected = false;
         }
 
-        public void OnGUI()
+        public static void ResetTimer()
         {
-            if (!ChatEnabled || Manager == null)
-            {
-                return;
-            }
-
-            SetGUIStyle();
-            GUI.SetNextControlName(GUI_CHAT_NAME);
-            chatMessage = GUI.TextField(new Rect(INPUT_MARGIN, Screen.height - INPUT_HEIGHT - INPUT_MARGIN, INPUT_WIDTH, INPUT_HEIGHT), chatMessage, CHAR_LIMIT);
-            GUI.FocusControl(GUI_CHAT_NAME);
-
-            if (Event.current.isKey)
-            {
-                switch (Event.current.keyCode)
-                {
-                    case KeyCode.Return:
-                        SendMessage();
-                        ChatEnabled = false;
-                        break;
-                    case KeyCode.Escape:
-                        Manager.HideLog();
-                        ChatEnabled = false;
-                        break;
-                }
-            }
+            timeLeftUntilAutoClose = PlayerChat.CHAT_VISIBILITY_TIME_LENGTH;
+            FreezeTime = false;
         }
 
-        private void SetGUIStyle()
+        private void LateUpdate()
```

Comment:
Change to use Update() if possible.

```suggestion
        private void Update()
```